sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/cryptography/CryptographyAsyncClient.java

```diff
@@ -949,6 +968,23 @@ private Mono<Boolean> ensureValidKeyAvailable() {
         }
     }
 
+    static boolean isCurveSupportedByRuntimeJavaVersion(KeyCurveName keyCurveName) {
+        String javaVersion = System.getProperty("java.version");
+
+        if (javaVersion.startsWith("1.")) {
+            javaVersion = javaVersion.substring(2, 3);
+        } else {
+            int period = javaVersion.indexOf(".");
+
+            if (period != -1) {
+                javaVersion = javaVersion.substring(0, period);
+            }
+        }
```

Comment:
Why are you repeating this in the caller? Looks exactly the same.

Reply:
I was trying to pre-emptively throw an error with a decent error message other than "Curve secp256k1 not supported". I'll remove this check and add this message on a catch block for when local client creation ultimately fails when trying to convert the given `JsonWebKey` to a `KeyPair`, which is where the error message I mentioned comes from.